design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,254 @@
+
+*Unintuitive command construction*
+
+When running `conjur variable value <variable_id>`, a secret value will be returned but when looking at just the command, it is unclear what the purpose of the command is. For a more straightforward approach, the command for displaying a secret value could be `conjur get <variable_id>` or `conjur show <variable_id>`.  
+
+Another example of uninuitive command construction would be trying to understand what the difference between `conjur variable value`  is and `conjur variable values` without using the `--help` flag or reading documentation.
+
+*Permitted edge cases*
+
+We are currently not blocking the chaining of illogical flagging. For example, `conjur policy load --delete --replace foo foo.yml` does not lead to an error.
+
+#### Python CLI UX
+
+The Python CLI has a more straightforward experience. For example, when setting and getting variables, the user will run
+
+```
+conjur-cli variable set my_var_id mysupersecretvar
+conjur-cli variable get my_var_id
+```
+
+For policy maintaince the outcome of the command is very clear. Unlike the Ruby/Go CLI, there is no chance of an unpredictable outcome because running `replace` and `delete`  would lead to an error.
+
+```bash
+conjur-cli policy apply root ./path_to_policy.yml
+conjur-cli policy replace root ./path_to_policy.yml
+conjur-cli policy delete root ./path_to_policy.yml
+```
+
+To use the Python CLI, users can either provide the information usually found in their `.netrc` / `.conjurrc` files as command line arguments. For example:
+
+```
+conjur-cli --insecure -l https://conjur.myorg.com ... -u sigal.sax -p 2ec....gv63 \
+  variable get foo/bar
+```
+
+|            | Deployment                                                   |
+| ---------- | ------------------------------------------------------------ |
+| Ruby CLI   | The Ruby CLI can be deployed in the following ways:<br />1. Deploy to Docker container (**not** applicable for our use-case)<br />2. Deploy with specific Ruby version (`gem install conjur-cli`)<br />Which means that the CLI is not plug-and-play and has prerequisites |
+| Go CLI     | The CLI is compiled down to a binary. A different binary is need to be compiled for each platform but no other requirements or installers are needed to launch the Go CLI. |
+| Python CLI | The Python CLI can deployed in the following ways:<br />1. Install CLI via PIP (Python **is** required on machine)<br />2. Runnable executable (Python **is** **not** required on the machine)<br />The CLI is delivered as a [standalone executable](https://github.com/cyberark/conjur-api-python3/releases/tag/v0.0.5), packed with the required runtime and dependencies. For this option there is no need for Python or other prerequisites to be installed on machine. |
```

Comment:
There may be dynamic library requirements for Linux but they should be minimal - I have not explored the support matrix much other than the most common Win/macOS/Linux permutations for those binaries.

Reply:
@sgnn7 is this the signed executables we spoke about?

Reply:
No - this is more baseline library requirement for a Linux system (e.g. glibc, libgz). Running `ldd` on the binary should give a list of minimal requirements for these. The latest release shows this as:
```
ldd ./conjur-py3-cli-linux 
        linux-vdso.so.1 (0x00007fff67590000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f8526758000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f852653a000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f8526379000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f8526768000)
```